api/src/main/java/au/org/aodn/nrmn/restapi/service/CorrectionService.java

```diff
@@ -0,0 +1,88 @@
+package au.org.aodn.nrmn.restapi.service;
+
+import au.org.aodn.nrmn.restapi.model.db.*;
+import au.org.aodn.nrmn.restapi.repository.ObservationRepository;
+import au.org.aodn.nrmn.restapi.repository.SurveyMethodRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Example;
+import org.springframework.stereotype.Service;
+
+import java.sql.Date;
+import java.sql.Time;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+public class CorrectionService {
+    @Autowired
+    private ObservationRepository observationRepository;
+    @Autowired
+    private SurveyMethodRepository surveyMethodRepository;
+
+    public Stream<StagedRow> convertSurveyToStagedRows(Survey survey) {
+        List<SurveyMethod> surveyMethods = surveyMethodRepository.findAll(Example.of(
+                SurveyMethod.builder().survey(survey).build()));
+        return surveyMethods.stream().map(this::convertSurveyMethodToStagedRow);
+    }
+
+    private StagedRow convertSurveyMethodToStagedRow(SurveyMethod surveyMethod) {
+        Survey survey = surveyMethod.getSurvey();
+        StagedRow.StagedRowBuilder builder = StagedRow.builder()
+                .depth(getStagedDepth(survey.getDepth(), survey.getSurveyNum()))
+                .siteNo(survey.getSite().getSiteCode())
+                .siteName(survey.getSite().getSiteName())
+                .date(getDateString(survey.getSurveyDate()))
+                .time(getTimeString(survey.getSurveyTime()))
+                .vis(toString(survey.getVisibility()))
+                .direction(survey.getDirection())
+                .block(toString(surveyMethod.getBlockNum()))
+                .method(toString(surveyMethod.getMethod().getMethodId()))
+                .code(surveyMethod.getSurveyNotDone() ? "SND" : null);
```

Comment:
what type of code is this?

Reply:
This is the species code column, but when they don't do a survey they use it to indicate  "survey not done", ie "SND" instead.

Reply:
yuk.  But assuming this reflects whats in the spreadsheet again?

Reply:
Yup, and if the survey has been done, then species code can be left blank since the species scientific name is the important thing for species.